cloud_level_testing/features/steps/tools.py

```diff
@@ -373,14 +384,92 @@ def delete_ports(context, port_ids: list = None):
                 context.logger.log_info(f"FAIL! Port {port_id} wasn't deleted")
 
 
+def delete_jumphosts(context, jumphost_ids: list = None):
+    """Delete jumphosts based on list of IDs or jumphost IDs in collector.
+
+    Args:
+        context: Behave context object.
+        jumphost_ids: List of jumphost IDs to delete, if None, collector jumphost IDs are used.
+    """
+    jumphost_ids = context.collector.jumphosts[:] if not jumphost_ids else jumphost_ids
+    delete_vms(context, jumphost_ids)
+
+
 def delete_all_test_resources(context):
-    """Delete all resources used in the feature run
+    """Delete all resources used in the feature run.
 
     Args:
         context: Behave context object
     """
     delete_vms(context)
+    delete_jumphosts(context)
     delete_ports(context)
     delete_subnets(context)
     delete_networks(context)
     delete_routers(context)
+
+
+def parse_ping_output(data: list[str], logger: Logger):
+    """Parse the outputs of the ping response test and print them to logger.
+
+    Args:
+        data: Ssh command stdout data.
+        logger: Default context logger.
+    """
+    start_time, end_time = get_timestamps(data[0])
+    calc_time = int(end_time) - int(start_time)
+    start_time = datetime.datetime.fromtimestamp(int(start_time))
+    end_time = datetime.datetime.fromtimestamp(int(end_time))
+    ping_data = data[1].split(sep="\n")
+    response_times = []
+    for ping_res in ping_data:
+        split_ping_res = ping_res.split()
+        timestamp = split_ping_res[0][1:11:1]
+        timestamp = datetime.datetime.fromtimestamp(int(timestamp))
+        if timestamp >= start_time and timestamp <= end_time:
+            response_times.append(float(split_ping_res[1].split(sep="=")[1]))
+    logger.log_info(f"Average response time is: {calc_average(response_times)} ms")
+    logger.log_info(f"Num of values: {len(response_times)}")
+    logger.log_info(f"Calc start: {start_time}")
+    logger.log_info(f"Calc ended: {end_time}")
+    logger.log_info(f"Calc took approx {calc_time}")
+
+
+def calc_average(values: list[float]) -> float:
+    sum = 0
+    for value in values:
+        sum = sum + value
+    return sum / len(values)
+
+
+def get_timestamps(data: str) -> tuple[str, str]:
```

Comment:
can you please let me know where the actual timestamps are from?

Reply:
Timestamps are from the VMs, they are taken before and after the calculation command.